src/module.c

```diff
@@ -118,9 +153,83 @@ static int _RenameGraphHandler(RedisModuleCtx *ctx, int type, const char *event,
 	return REDISMODULE_OK;
 }
 
-static void RegisterForkHooks(RedisModuleCtx *ctx) {
+// Module level function for RDB start save event - Create the meta keys for each graph.
+static void _CreateKeySpaceMetaKeys(RedisModuleCtx *ctx) {
+	uint graphs_in_keyspace_count = array_len(graphs_in_keyspace);
+	for(uint i = 0; i < graphs_in_keyspace_count; i ++) {
+		GraphContext_CreateMetaKeys(ctx, graphs_in_keyspace[i]);
+	}
+}
+
+// Module level function for RDB start end or load end event - Delete the meta keys for each graph.
+static void _ClearKeySpaceMetaKeys(RedisModuleCtx *ctx) {
+	uint graphs_in_keyspace_count = array_len(graphs_in_keyspace);
+	for(uint i = 0; i < graphs_in_keyspace_count; i ++) {
+		GraphContext_DeleteMetaKeys(ctx, graphs_in_keyspace[i]);
+	}
+}
+
+// Delete all meta keys before the actual flush, as the flush may be out of order
+static void _FlushDBHandler(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent,
+							void *data) {
+	if(eid.id == REDISMODULE_EVENT_FLUSHDB && subevent == REDISMODULE_SUBEVENT_FLUSHDB_START) {
+		_ClearKeySpaceMetaKeys(ctx);
+	}
+}
+
+// Checks if the event is persistence start event.
+static bool _IsEventPersistenceStart(RedisModuleEvent eid, uint64_t subevent) {
+	return eid.id == REDISMODULE_EVENT_PERSISTENCE  &&
+		   // Normal RDB.
+		   (subevent == REDISMODULE_SUBEVENT_PERSISTENCE_RDB_START ||
```

Comment:
I suggest having the comment and the relevant flag on the same line (for all these functions).